sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,744 @@
+#else
+// Required for sleep(unsigned int)
+#include <unistd.h>
+#endif
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+#include <azure/iot/az_iot_hub_client.h>
+
+// TODO: #564 - Remove the use of the _az_cfh.h header in samples.
+//              Note: this is required to work-around MQTTClient.h as well as az_span init issues.
+#include <azure/core/_az_cfg.h>
+
+// DO NOT MODIFY: Device ID Environment Variable Name
+#define ENV_DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// DO NOT MODIFY: IoT Hub Hostname Environment Variable Name
+#define ENV_IOT_HUB_HOSTNAME "AZ_IOT_HUB_HOSTNAME"
+
+// DO NOT MODIFY: The path to a PEM file containing the device certificate and
+// key as well as any intermediate certificates chaining to an uploaded group certificate.
+#define ENV_DEVICE_X509_CERT_PEM_FILE "AZ_IOT_DEVICE_X509_CERT_PEM_FILE"
+
+// DO NOT MODIFY: the path to a PEM file containing the server trusted CA
+// This is usually not needed on Linux or Mac but needs to be set on Windows.
+#define ENV_DEVICE_X509_TRUST_PEM_FILE "AZ_IOT_DEVICE_X509_TRUST_PEM_FILE"
+
+#define TIMEOUT_MQTT_DISCONNECT_MS (10 * 1000)
+#define TELEMETRY_SEND_INTERVAL 1
+#define NUMBER_OF_MESSAGES 5
+#define DEAFULT_START_TEMP_CELSIUS 22
+
+static const uint8_t null_terminator = '\0';
+static char boot_time_str[32];
+static az_span boot_time_span;
+
+// IoT Hub Connection Values
+static az_iot_hub_client client;
+static char device_id[64];
+static char iot_hub_hostname[128];
+static char x509_cert_pem_file[512];
+static char x509_trust_pem_file[256];
+static const az_span model_id = AZ_SPAN_LITERAL_FROM_STR("dtmi:com:example:Thermostat;1");
+
+// MQTT Client Values
+static MQTTClient mqtt_client;
+static char mqtt_client_id[128];
+static char mqtt_username[256];
+static char mqtt_endpoint[128];
+static az_span mqtt_url_prefix = AZ_SPAN_LITERAL_FROM_STR("ssl://");
+static az_span mqtt_url_suffix = AZ_SPAN_LITERAL_FROM_STR(":8883");
+
+// IoT Hub Telemetry Values
+char telemetry_topic[128];
+static const az_span telemetry_name = AZ_SPAN_LITERAL_FROM_STR("temperature");
+static char telemetry_payload[32];
+
+// IoT Hub Direct Methods Values
+static char methods_response_topic[128];
+static const az_span method_payload_value = AZ_SPAN_LITERAL_FROM_STR("since");
+static const az_span report_method_name_span = AZ_SPAN_LITERAL_FROM_STR("getMaxMinReport");
+static const az_span report_max_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("maxTemp");
+static const az_span report_min_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("minTemp");
+static const az_span report_avg_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("avgTemp");
+static const az_span report_start_time_name_span = AZ_SPAN_LITERAL_FROM_STR("startTime");
+static const az_span report_end_time_name_span = AZ_SPAN_LITERAL_FROM_STR("endTime");
+static char end_time_buffer[32];
+static char methods_response_payload[256];
+
+// IoT Hub Twin Values
+static char reported_property_topic[128];
```

Comment:
fold all the topic buffer into one instance?

Reply:
we might be able to but it's an optimization at the cost of clarity. I'll keep as is unless others are passionate about it.